lib/SearchAndSort/components/DateRangeFilter/date-validations.js

```diff
@@ -0,0 +1,42 @@
+import moment from 'moment';
+
+export const isDateValid = (date, dateFormat) => {
+  const momentDateWrapper = moment(date, dateFormat, true);
+
+  return momentDateWrapper.isValid();
+};
+
+export const validateDateRange = (dateRange, dateFormat) => {
```

Comment:
This is really nicely implemented. It's clear and easy to follow, and makes it easy to tell exactly which conditions failed. Nice work!

Reply:
Thank you @zburke!